assets/src/apps/delivery/store/features/groups/actions/deck.ts

```diff
@@ -126,11 +152,39 @@ export const initializeActivity = createAsyncThunk(
       };
       sessionOps.push(targetVisitTimeStampOp);
     }
-
     // init state is always "local" but the parts may come from parent layers
     // in that case they actually need to be written to the parent layer values
     const initState = currentActivity?.content?.custom?.facts || [];
     const arrInitFacts: string[] = [];
+
+    const currentSequenceLineage = getSequenceLineage(sequence, currentSequenceId);
+    const isQuestionBankActivity = currentSequenceLineage.some((s) => s.custom.isBank);
+
+    const quetionBankResetScript: ApplyStateOperation[] = [];
+    if (isQuestionBankActivity && currentActivityTree) {
+      // question bank children cannot be nested, so the parent must be the bank
+      const parentBank = currentSequenceLineage[currentSequenceLineage.length - 1];
+      // in delivery activity id is sequence id
+      const bankActivity = selectActivityById(rootState, parentBank.custom.sequenceId);
+      if (bankActivity) {
+        currentActivityTree.forEach((activity) => {
```

Comment:
I dont understand, doing the entire activity tree seems the opposite of not resetting layers, all of the layer ancestors will be in the tree. before I was *only* doing the question bank, not even the current activity (though the current activity should be doing it on its own during init anyway).

Reply:
But we do have condition there which only reset variables of question bank activity and question bank layer.

`if (activity.id === parentBank.custom.layerRef || activity.id === bankActivity.id) {}`

Also, the reason for that is, layer don't get rendered as screen so the components that are on layer, they never get called if we do current activity reset